src/state.py

```diff
@@ -42,14 +45,19 @@ class Configuration(pydantic.BaseModel):
         hostname: The hostname for the virtual host for this set of configuration.
         path: The path for this set of configuration.
         backends: The backends for this set of configuration.
-        protocol: The protocol to request the backends with. Can be http or
-            https.
+        protocol: The protocol to request the backends with. Can be http or https.
+        fail_timeout: The time to wait before using a backend after failure.
+        backends_path: The path to request the backends.
+        proxy_cache_valid: The cache valid duration.
     """
 
     hostname: typing.Annotated[str, pydantic.StringConstraints(min_length=1)]
     path: typing.Annotated[str, pydantic.StringConstraints(min_length=1)]
     backends: tuple[pydantic.IPvAnyAddress, ...]
     protocol: Protocol
+    fail_timeout: typing.Annotated[str, pydantic.StringConstraints(min_length=1)]
```

Comment:
Just an alternative, you can also call the validator in the `Annotated`
ex: 
```python
fail_timeout: typing.Annotated[str, pydantic.StringConstraints(min_length=1), BeforeValidator(check_nginx_time_str)]
```

https://docs.pydantic.dev/latest/concepts/validators/#annotated-validators